src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ThrowingExceptions/WalkerForGraphAnalyzer.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ThrowingExceptions
+{
+    internal class WalkerForGraphAnalyzer : WalkerBase
+    {
+        private readonly DiagnosticDescriptor _descriptor;
+
+        public WalkerForGraphAnalyzer(SymbolAnalysisContext context, PXContext pxContext, DiagnosticDescriptor descriptor)
+                : base(context, pxContext)
+        {
+            descriptor.ThrowOnNull(nameof(descriptor));
+
+            _descriptor = descriptor;
+        }
+
+        public override void VisitThrowStatement(ThrowStatementSyntax node)
+        {
+            ThrowIfCancellationRequested();
+
+            if (IsPXSetupNotEnteredException(node))
```

Comment:
@metallmonster please consider enhancing of `IsPXSetupNotEnteredException` method.
Currently it starts like this:
```
if (node == null || !(node.Expression is ObjectCreationExpressionSyntax objCreationSyntax) ||
                objCreationSyntax.Type == null)
            {
                return false;
            }
```
This leaves only expressions like:
`throw new MyExceptionType();`
It would be good to support also variables stored in exceptions:
```
var exception = new MyExceptionType();
PXTrace.WriteError(exception);
throw exception ;
```